src/pudl/transform/ferc1.py

```diff
@@ -484,11 +634,11 @@ def process_dbf(self, raw_dbf: pd.DataFrame) -> pd.DataFrame:
         return (
             raw_dbf.drop_duplicates()
             .pipe(self.drop_footnote_columns_dbf)
+            .pipe(self.align_row_numbers_dbf)
             # Note: in this rename_columns we have to pass in params, since we're using
             # the inherited method, with param specific to the child class.
             .pipe(self.rename_columns, params=self.params.rename_columns_ferc1.dbf)
             .pipe(self.assign_record_id, source_ferc1=Ferc1Source.DBF)
-            .pipe(self.align_row_numbers_dbf)
             .pipe(self.drop_unused_original_columns_dbf)
```

Comment:
move this up in the stack bc we are wanting to to the things that require knowing what the name of the row name columns actually is before the rename step so we don't have to propagate the column name across many transforms

Reply:
Wow, are you surprised that this works without any other tweaks?